server/container/guice/cassandra-guice/src/test/java/org/apache/james/CassandraConfTest.java

```diff
@@ -0,0 +1,78 @@
+package org.apache.james;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.nio.ByteBuffer;
+import java.nio.channels.SocketChannel;
+import java.nio.charset.Charset;
+
+import org.apache.commons.configuration.PropertiesConfiguration;
+import org.apache.james.modules.mailbox.CassandraSessionConfiguration;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import com.google.common.base.Joiner;
+
+public class CassandraConfTest {
+
+    private static final int IMAP_PORT = 1143;
+    private static final int CASSANDRA_PORT = 9042;
+	private GuiceJamesServerImpl jamesServer;
+	private SocketChannel socketChannel;
+
+
+    private PropertiesConfiguration getCassandraConfigurationForDocker(String... ips) {
+        PropertiesConfiguration configuration = new PropertiesConfiguration();
+        
+        configuration.addProperty("cassandra.ip", Joiner.on(',').join(ips));
+        configuration.addProperty("cassandra.port", CASSANDRA_PORT);
+        configuration.addProperty("cassandra.keyspace", "apache_james");
+        configuration.addProperty("cassandra.replication.factor", 1);
+        configuration.addProperty("cassandra.retryConnection.maxRetries", 10);
+        configuration.addProperty("cassandra.retryConnection", 5000);
+
+        return configuration;
+    }
+    private final DockerCassandraRule dockerCassandraRule = new DockerCassandraRule();
+	@Rule public CassandraJmapTestRule cassandraJmapTestRule = new CassandraJmapTestRule(dockerCassandraRule, new EmbeddedElasticSearchRule());
+
+	@Before
+	public void setUp() throws IOException {
+	    socketChannel = SocketChannel.open();
+	}
+
+	@After
+	public void after() throws IOException {
+	    socketChannel.close();
+	}
+
+	@Test
+    public void serverShouldStartServiceAreReachable() throws Exception {
+	    jamesServer = cassandraJmapTestRule.jmapServer();
+	    jamesServer.start();
+        socketChannel.connect(new InetSocketAddress("127.0.0.1", IMAP_PORT));
+        assertThat(getServerConnectionResponse(socketChannel)).startsWith("* OK JAMES IMAP4rev1 Server");
+    }
+	
+	@Test
+	public void serverShouldStartWhenOneCassandraNodeIsUnreachable() throws Exception {
+	    PropertiesConfiguration configuration = getCassandraConfigurationForDocker("10.2.3.42", dockerCassandraRule.getIp());
```

Comment:
The thing is if we do not put an IP but a hostname Cassandra's driver try to resolve it and fail without trying the second ip/hostname. That the reason why we put a random IP.